docs/design/trimmed-assemblies.md

```diff
@@ -0,0 +1,183 @@
+
+### Examples
+
+This shows how a developer could turn on aggressive trimming for framework assemblies (which are defined to be `IsTrimmable` by the SDK):
+
+```xml
+<PropertyGroup>
+  <TrimMode>link</TrimMode>
+</PropertyGroup>
+```
+
+This shows how Blazor (or a developer) could hook into the build to opt assemblies into different levels of trimming based on the filename:
+
+```xml
+<Target Name="PrepareForBlazorILLink"
+        BeforeTargets="PrepareForILLink">
+  <PropertyGroup>
+    <!-- Set the default TrimMode for IsTrimmable assemblies -->
+    <TrimMode>link</TrimMode>
+  </PropertyGroup>
+  <ItemGroup>
+    <ManagedAssemblyToLink Condition="'$([System.String]::Copy('%(ManagedAssemblyToLink.Filename)').StartsWith('Microsoft.AspNetCore.'))">
+      <!-- Trim these assemblies using the global TrimMode -->
+      <IsTrimmable>true</IsTrimmable>
+    </ManagedAssemblyToLink>
+    <ManagedAssemblyToLink Condition="'$([System.String]::Copy('%(ManagedAssemblyToLink.Filename)').StartsWith('ThirdPartyAssembly.'))">
+      <!-- Trim these assemblies with assembly-level trimming. Implies IsTrimmable. -->
+      <TrimMode>copyused</TrimMode>
+    </ManagedAssemblyToLink>
+  </ItemGroup>
+</Target>
+```
+
+### Other options
+
+.NET 5 introduced a host of additional SDK options that map directly to the underlying illink options. The full list is documented at https://docs.microsoft.com/en-us/dotnet/core/deploying/trimming-options.
+
+## .NET 6
+
+### `IsTrimmable` `AssemblyMetadataAttribute`
+
+An assembly-level `AssemblyMetadataAttribute` may be placed on an assembly to indicate that it should be trimmed:
+
+```csharp
+[AssemblyMetadataAttribute("IsTrimmable", "true")]
+```
+
+The behavior is the same as the `IsTrimmable` MSBuild metadata, so that:
+- Assemblies with this attribute are trimmed with the global `TrimMode`
+- Assemblies without this attribute are rooted, and given the `copy` action
+
+The only understood value is `true`. Adding `[AssemblyMetadataAttribute("IsTrimmable", "false")]` will have no effect on the linker's behavior, because unattributed assemblies are assumed not to be trimmable by default. We will issue a warning in this case, to discourage misleading use of the attribute.
+
+If `IsTrimmable` MSBuild metadata is set for an assembly, this overrides the `IsTrimmable` attribute. This allows a developer to opt an assembly into trimming even if it does not have the attribute, or to disable trimming of an assembly that has the attribute.
+
+Instead of using `IsTrimmable` metadata in the SDK to trim netcoreapp by default, we will ship netcoreapp assemblies and OOB assemblies that are conceptually part of the BCL with `IsTrimmable` `AssemblyMetadataAttribute`, and not set `IsTrimmable` MSBuild metadata in the SDK. The MSBuild metadata will now be used only as a developer opt-in or opt-out.
```

Comment:
```suggestion
Instead of using `IsTrimmable` metadata in the SDK to control trimmable assemblies, we will move to a model where all trimmable SDKs assemblies are built with `[AssemblyMetadataAttribute("IsTrimmable", "true")]`